openwisp_monitoring/monitoring/tests/test_db_creation.py

```diff
@@ -0,0 +1,25 @@
+from unittest.mock import patch
+
+from django.test import TestCase
+from influxdb.exceptions import InfluxDBClientError
+
+from .. import settings as app_settings
+from ..utils import create_database, get_db
+
+
+def mock_create_database(self, db):
+    raise(InfluxDBClientError('database not created'))
+
+
+class TestDatabase(TestCase):
+    ORIGINAL_DB = app_settings.INFLUXDB_DATABASE
+    TEST_DB = 'test_db'
+
+    @patch('openwisp_monitoring.monitoring.utils.warn_and_delay')
+    @patch('influxdb.client.InfluxDBClient.create_database', mock_create_database)
+    def test_check_retry(self, mock):
+        setattr(app_settings, 'INFLUXDB_DATABASE', self.TEST_DB)
```

Comment:
can you try mocking this as well?

`@patch('openwisp_monitoring.monitoring.app_settings.INFLUXDB_DATABASE', 'test_db')`